src/test/java/com/codeborne/selenide/impl/ScreenShotLaboratoryTest.java

```diff
@@ -96,6 +101,17 @@ void collectsAllScreenshots() {
       .hasToString("12356789.4");
   }
 
+  @Test
+  void canFormatReportsURLWithSpaces() {
+    Driver driver = new DriverStub(new SelenideConfig().reportsUrl("http://ci.org/with space/"),
```

Comment:
Is `"http://ci.org/with%20space/"` legal to be used in this context? Should it be different from `"http://ci.org/with space/"` or should it be the same?

Would you please add a test for `"http://ci.org/with%20space/"` as well?

Reply:
Yes, it is a rare case when URL can contain space characters or when there is a folder on CI with spaces. But the main goal was to escape special characters when, for example, the test method has spaces, like described there #1072

Reply:
I see, however, it looks weird to concatenate strings with different `encoding` nature.

In other words, `reportsURL` is already an encoded URL, however, file path is not encoded.

So I think the approach should be **skip** encoding for `reportsURL` and **perform** encoding of the path added.

That, however, keeps the question of how the URL should look like if `reportsURL` is like `https://host:port/get?param=value`

Reply:
The specification says that every part of URL should be encoded separately. 
I think we can assume that `reportsURL` is already encoded. 

So, the resulting code would be 
`reportsURL + "?" + encode(paramName) + "=" + encode(paramValue)`

See https://www.talisman.org/~erlkonig/misc/lunatech%5Ewhat-every-webdev-must-know-about-url-encoding/#DonotconstructURLswithoutencodingeachpart

Reply:
I am really confused about this query in base reports URL. From my understanding, they are needed to send some parameters to a specific resource on the server. So as @vlsi said, where should we append it in the resulted URL?

For example:
if someone provides `http://ci.mycompany.com/job/666/artifact?some=value` and we append path `build/reports/tests/screenshot.html` how should look the resulting URL? 
Are these correct?
`http://ci.mycompany.com/job/666/artifact?some=value/build/reports/tests/screenshot.png` or
`http://ci.mycompany.com/job/666/artifact/build/reports/tests/screenshot.png?some=value`

At least (if without query) I can push another version where I encode the `path` without `reportUrl`

Reply:
I assume that `reportsUrl` does NOT contain any parameters. 
* `http://ci.mycompany.com/job/666/artifact?some=value` is *invalid* value
* `http://ci.mycompany.com/job/666/artifact/path%20can%20contain%20spaces/` is a *valid* value.